libfreerdp/core/window.c

```diff
@@ -294,6 +294,7 @@ BOOL update_read_window_state_order(wStream* s, WINDOW_ORDER_INFO* orderInfo, WI
 
 		Stream_Read_UINT16(s, windowState->numWindowRects); /* numWindowRects (2 bytes) */
 
+		free(windowState->windowRects);
 		size = sizeof(RECTANGLE_16) * windowState->numWindowRects;
 		windowState->windowRects = (RECTANGLE_16*) malloc(size);
```

Comment:
I guess we can do realloc() here instead of free() + malloc(). You have to be careful to handle realloc() failure, but I think this field will be eventually freed, so we can simply return here if realloc fails, just as we do now for malloc().